### PR TITLE
WIP: Feature/AC pulses

### PR DESCRIPTION
**Pull request by dieris**
To be tested. Also, where should we input `sampling_rate`?

### CODE REVIEW

File: src/pulses.jl [hunk 1]
```diff
@@ -84,23 +84,39 @@ end
...
-		q -> Y90m(q),
-		q -> Z90(q),
-		q -> Z(q),
-		q -> Z90m(q)
-	]
-
-	return pulses[num](q)
-
+function AC(q::Qubit, num; sampling_rate = 1.2e9)
+	if 0<num<=12
```
Comment (marcusps) [inline src/pulses.jl]: You are using `num` to index into a vector, but `num` can be 0.  Did you intent to have 0 be a zero length identity?

Comment (dieris) [reply src/pulses.jl]: I think that it was the original plan, but then I forgot to include the case for 0. I'll add it

**Comment by caryan** [comment]
Since `sampling_rate` is just used to calculate the nutation frequency and passed through to the shape function we could defer it till then when we will be given the sampling rate by passing the `Xp` Pulse object

**Comment by caryan** [comment]
For the arb_axis_drag we also need to add the frame changes.

**Comment by dieris** [comment]
I'm confused, is there supposed to be a frame change at the end of the arb_axis_drag pulse? How is it accomplished [here](https://github.com/BBN-Q/QGL/blob/d8af924a7c87c7974f547850ac58286dcad41d0c/QGL/PulseShapes.py#L221)?

**Comment by caryan** [comment]
Seems like unnecessary copy-paste bonus that we calculate the frame change in `PulseShapes.py` but the point is that we have a frame change in the `Pulse` definition [here](https://github.com/BBN-Q/QGL/blob/d8af924a7c87c7974f547850ac58286dcad41d0c/QGL/PulsePrimitives.py#L308).  From QGL.jl we need to return a `PulseBlock` with the `ZPulse`.

**Comment by caryan** [comment]
At @dieris  I'm just going to merge this so it doesn't get left dangling.  We can file and fix issues when we use the pulses.

**Pull request merged**